### ISSUE TITLE
"NoneType object not callable” error when calling sfpreset_name()

### ISSUE THREAD

**Issue by erixoff** [nwhitehead/pyfluidsynth#35]
Hello,
First of all, thanx very much for bringing PyFluidSynth.

I recently installed PyFluidSynth in a 32bit Windows environment, and managed to find a 32bit version of FluidSynth (wich is not so easy to find, unless I rebuild it) in a form of libfluidsynth-1.dll
I execute the first test listed on PyFluidSynth GitHub homepage and it works well.
Then I tried to call the following services and they fail with “NoneType object not callable” error :
- fs.channel_info()
- sfpreset_name(sfid, 0, 0)

Same issue with the example with the FluidSynth sequencer.

  _File "C:\Users\xxx\MyApp\Python3.7_32bits\lib\site-packages\fluidsynth.py", line 607, in channel_info
    presetname = self.sfpreset_name(sfontid, banknum, prognum)
  File "C:\Users\xxx\MyApp\Python3.7_32bits\lib\site-packages\fluidsynth.py", line 624, in sfpreset_name
    preset=fluid_sfont_get_preset(sfont, bank, prog)
TypeError: 'NoneType' object is not callable_
 
Did I miss something ?

Thanx in advance

Regards

**Comment by albedozero** [comment]
Sorry I forgot to come back to this for a couple weeks. Did you copy *all* the DLL's from the 32bit release into your project root? They're all needed for fluidsynth to work.

**Comment by erixoff** [comment]
No problemo, you're back; that matters 👍 
I copied all the DLLs and now, it works; I have sound and just the following error messages, but it runs:
_fluidsynth: error: not enough MIDI in devices found. Expected:1 found:0
fluidsynth: error: Device "default" does not exists
fluidsynth: warning: sequencer: Usage of the system timer has been deprecated!_
Thanx very much !

**Comment by albedozero** [comment]
> _fluidsynth: error: not enough MIDI in devices found. Expected:1 found:0_
>_fluidsynth: error: Device "default" does not exists_

This is because `midi.autoconnect` is set to `1` and you didn't have a MIDI controller connected or it wasn't detected because it's being used by something else. `midi.autoconnect` should be `0` by default - if you're not planning to play notes from a keyboard you don't need it.

> _fluidsynth: warning: sequencer: Usage of the system timer has been deprecated!_

Either `player.timing-source` is set to `system`, or maybe your script is rendering MIDI files to WAV files? If the latter, system timing is your only option and it's fine anyway. If you want fluidsynth to output audio to speakers, sample timing is probably better.

TLDR if your script is doing what you want, no worries! 😀

**Comment by erixoff** [comment]
Thanx very much for your answers (yes, I don't have MIDI keyboard connected, and script runs fine), for your support, and for having taken your time for my request.
Good day
